### PR TITLE
Improving documentation and support for deep-framework

### PR DESCRIPTION
**Pull request by eistrati**
- Adding documentation on how to contribute to this project
- Adding unit testing and better support for ES6

### CODE REVIEW

File: src/deep-log/lib/Driver/ConsoleDriver.js [hunk 1]
```diff
@@ -20,8 +50,71 @@ export class ConsoleDriver extends AbstractDriver {
    * @param {*} context
    */
```
Comment (devs-deep) [inline src/deep-log/lib/Driver/ConsoleDriver.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (10)](https://www.codacy.com/app/MitocGroup/deep-framework/file/1329751725/issues/source?bid=2888470&fileBranchId=2957139#l52)

**Comment by eistrati** [comment]
Please fix the issues reported by Travis, Codacy and Hound

File: src/deep-db/lib/Local/Driver/LocalDynamo.js [hunk 2]
```diff
@@ -35,21 +52,32 @@ export class LocalDynamo extends PathAwareDriver {
...
+    fse.ensureDirSync(this.path);
+
     this._options.dir = this.path;
 
-    this._process = LocalDynamoServer.launch(this._options, this.port);
+    try {
+      this._process = LocalDynamoServer.launch(this._options, this.port);
+    } catch (error) {
+      this._pickUpOldInstance
+        ? cb(null)
```
Comment (devs-deep) [inline src/deep-db/lib/Local/Driver/LocalDynamo.js]: Codacy Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/MitocGroup/deep-framework/file/1339245290/issues/source?bid=2888470&fileBranchId=2957139#l65)

File: src/deep-log/lib/Driver/ConsoleDriver.js (hunk 1 above)
Comment (devs-deep) [inline src/deep-log/lib/Driver/ConsoleDriver.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (10)](https://www.codacy.com/app/MitocGroup/deep-framework/file/1339245341/issues/source?bid=2888470&fileBranchId=2957139#l52)

**Pull request closed without merging**